app/controllers/indicators_controller.rb

```diff
@@ -72,6 +73,10 @@ def set_indicator_data
     @summary_value = calculations[:summary_value]
   end
 
+  def set_coin_stats
+    @coin_stats = Rails.cache.read("indicators/#{@coin.slug}:stats")
```

Comment:
Dumb question - what happens if the cache isn't populated?

Would recommend lazy loading the `CoinService` call here if the cache is empty just in case, that's generally a best practice when it comes to caching logic - the only exception is if the job takes entirely too long for an HTTP request to deal with, in which case there should be reasonable default value returned instead.

Reply:
It would just omit the section. Will add the call to coin service.